besu/src/main/java/org/hyperledger/besu/cli/subcommands/PublicKeySubCommand.java

```diff
@@ -62,25 +63,22 @@
   private CommandSpec spec; // Picocli injects reference to command spec
 
   private final PrintStream out;
-  private final KeyLoader keyLoader;
 
-  public PublicKeySubCommand(final PrintStream out, final KeyLoader keyLoader) {
+  public PublicKeySubCommand(final PrintStream out) {
     this.out = out;
-    this.keyLoader = keyLoader;
   }
 
   @Override
   public void run() {
     spec.commandLine().usage(out);
   }
 
-  private Optional<KeyPair> getKeyPair() {
-    try {
-      return Optional.of(keyLoader.load(parentCommand.nodePrivateKeyFile()));
-    } catch (IOException e) {
-      LOG.error("An error occurred while trying to read the private key", e);
-      return Optional.empty();
-    }
+  private BesuController<?> createBesuController() {
+    return parentCommand.buildController();
+  }
+
+  private Optional<NodeKey> getNodeKey() {
+    return parentCommand.getNodeKey();
```

Comment:
can we get this from the BesuController rather than the BesuCommand?

Reply:
done.